src/openforms/submissions/form_logic.py

```diff
@@ -136,8 +142,13 @@ def evaluate_form_logic(
                     variable_key = action["variable"]
                     updated_step_data[variable_key] = new_value
                     updated_submission_data = {**data, **updated_step_data}
-
+        else:
+            evaluated_rules.append({"rule": rule, "trigger": False})
     step.data = DirtyData(updated_step_data)
+    if evaluated_rules:
+        logevent.submission_logic_evaluated(
+            submission, evaluated_rules, updated_submission_data
+        )
```

Comment:
I'd consider always calling this and doing the check `if evaluated_rules` inside of `logevent.submission_logic_evaluated`